workers/typescript/src/workflows/kitchen_sink.ts

```diff
@@ -0,0 +1,295 @@
+import { google, temporal } from '../protos/root';
+import {
+  ActivityCancellationType as WFActivityCancellationType,
+  ApplicationFailure,
+  CancellationScope,
+  ChildWorkflowHandle,
+  ChildWorkflowOptions,
+  condition,
+  continueAsNew,
+  defineQuery,
+  defineSignal,
+  defineUpdate,
+  deprecatePatch,
+  isCancellation,
+  patched,
+  scheduleActivity,
+  scheduleLocalActivity,
+  setHandler,
+  sleep,
+  startChild,
+  upsertSearchAttributes,
+  Workflow
+} from '@temporalio/workflow';
+import {
+  ActivityOptions,
+  decompileRetryPolicy,
+  LocalActivityOptions,
+  SearchAttributes
+} from '@temporalio/common';
+import Long from 'long';
+import WorkflowInput = temporal.omes.kitchen_sink.WorkflowInput;
+import WorkflowState = temporal.omes.kitchen_sink.WorkflowState;
+import Payload = temporal.api.common.v1.Payload;
+import DoSignalActions = temporal.omes.kitchen_sink.DoSignal.DoSignalActions;
+import IActionSet = temporal.omes.kitchen_sink.IActionSet;
+import DoActionsUpdate = temporal.omes.kitchen_sink.DoActionsUpdate;
+import IAction = temporal.omes.kitchen_sink.IAction;
+import IPayload = temporal.api.common.v1.IPayload;
+import IAwaitableChoice = temporal.omes.kitchen_sink.IAwaitableChoice;
+import IExecuteActivityAction = temporal.omes.kitchen_sink.IExecuteActivityAction;
+import ActivityCancellationType = temporal.omes.kitchen_sink.ActivityCancellationType;
+import IDuration = google.protobuf.IDuration;
+import IWorkflowState = temporal.omes.kitchen_sink.IWorkflowState;
+
+const reportStateQuery = defineQuery<IWorkflowState, [Payload]>('report_state');
+const actionsSignal = defineSignal<[DoSignalActions]>('do_actions_signal');
+const actionsUpdate = defineUpdate<IPayload | undefined, [DoActionsUpdate]>('do_actions_update');
+
+export async function kitchenSink(input: WorkflowInput | undefined): Promise<IPayload | undefined> {
+  let workflowState: IWorkflowState = WorkflowState.create();
+  const actionsQueue = new Array<IActionSet>();
+
+  async function handleActionSet(actions: IActionSet): Promise<IPayload | null> {
```

Comment:
nit: prefer `undefined` over `null`, `undefined` payload converter is the equivalent of nil payload converter in sane languages that have somewhat better null representation.

It also will make it compatible with `actionsUpdate`'s return value.